staging/src/k8s.io/apiserver/pkg/storageversion/manager.go

```diff
@@ -35,6 +35,11 @@ import (
 type ResourceInfo struct {
 	GroupResource schema.GroupResource
 
+	// Optional. Overwrites the group name when creating the StorageVersion object.
+	// Used by the core group to make the group name more readable and a valid DNS
+	// subdomain.
+	OverwriteGroupName string
```

Comment:
Can we just write the ResourceInfo.GroupResource.Group when constructing the resourceInfo?

Reply:
If we change `ResourceInfo.GroupResource.Group`, the `EquivalentResourceMapper` won't recognize the new group name; also the name in `PendingUpdate` will be different, meaning we need to add special handling in both places. 

Essentially we just want to change the name of the StorageVersion object in etcd, the rest places remain unchanged:
- etcd: core.pods -- this makes the name validation happy, and the name more human readable (compared to `.pods`) for debugging purpose
- what `EquivalentResourceMapper` understands: GroupResource{Group: "", Resource: "pods"}
- `managedStatus` in `PendingUpdate`: GroupResource{Group: "", Resource: "pods"}
- `RequestInfo` we see in the filter: GroupResource{Group: "", Resource: "pods"}, which matches `PendingUpdate`

Alternatively we can remove the name validation in apiserver if we think the current implementation is not clean enough.

Reply:
> Alternatively we can remove the name validation in apiserver

Talked offline, in the [API reference doc](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#endpoints-v1-core) the group name is "core".